<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from new_gurdy_drive.urdf.xacro     | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="new_gurdy_drive" xmlns:xacro="http://www.ros.org/wiki/xacro">
  <gazebo>
    <plugin filename="libgazebo_ros_control.so" name="gazebo_ros_control">
      <robotNamespace>/new_gurdy</robotNamespace>
    </plugin>
  </gazebo>
  <!--MACROS-->
  <!-- Writing the xacro of transmissions -->
  <material name="blue">
    <color rgba="0 0 0.8 1"/>
  </material>
  <material name="red">
    <color rgba="0.8 0 0 1"/>
  </material>
  <material name="green">
    <color rgba="0 0.8 0 1"/>
  </material>
  <!-- *** Link Definitions *** -->
  <link name="base_link">
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.01 0.01 0.01"/>
      </geometry>
    </collision>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0.01"/>
      <geometry>
        <box size="0.01 0.01 0.01"/>
      </geometry>
    </visual>
  </link>
  <link name="head_link">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0.02"/>
      <mass value="0.01"/>
      <inertia ixx="7.58333333333e-07" ixy="0.0" ixz="0.0" iyy="7.58333333333e-07" iyz="0.0" izz="1.25e-06"/>
    </inertial>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0.02"/>
      <geometry>
        <cylinder length="0.04" radius="0.05"/>
      </geometry>
    </collision>
    <visual>
      <origin rpy="0.0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://gurdy_description/models/gurdy/meshes/gurdy_head_v2.dae"/>
      </geometry>
    </visual>
  </link>
  <gazebo reference="head_link">
    <mu1>10.0</mu1>
    <mu2>10.0</mu2>
  </gazebo>
  <joint name="fixed" type="fixed">
    <parent link="base_link"/>
    <child link="head_link"/>
  </joint>
  <link name="antenna_link">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0.05"/>
      <mass value="0.005"/>
      <inertia ixx="3.01322916667e-06" ixy="0.0" ixz="0.0" iyy="3.01322916667e-06" iyz="0.0" izz="5.625e-09"/>
    </inertial>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0.05"/>
      <geometry>
        <cylinder length="0.085" radius="0.0015"/>
      </geometry>
    </collision>
    <visual>
      <origin rpy="0.0 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder length="0.085" radius="0.0015"/>
      </geometry>
      <material name="green"/>
    </visual>
  </link>
  <joint name="antenna_joint" type="fixed">
    <parent link="head_link"/>
    <child link="antenna_link"/>
    <origin rpy="0.0 0.785 0.0" xyz="0.0 0.0 0.051"/>
  </joint>
  <!--Assign number values to the links/transmission -->
  <link name="aux_1_link">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0.0285"/>
      <mass value="0.01"/>
      <inertia ixx="2.64583333333e-08" ixy="0.0" ixz="0.0" iyy="2.64583333333e-08" iyz="0.0" izz="1.125e-08"/>
    </inertial>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0.00285"/>
      <geometry>
        <cylinder length="0.005" radius="0.0015"/>
      </geometry>
    </collision>
    <visual>
      <origin rpy="0.0 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder length="0.005" radius="0.0015"/>
      </geometry>
      <material name="blue"/>
    </visual>
  </link>
  <joint name="head_aux_1_joint" type="revolute">
    <parent link="head_link"/>
    <child link="aux_1_link"/>
    <origin rpy="3.145159 0 0.523599" xyz="-0.02165 -0.0125 -0.008"/>
    <limit effort="1.0" lower="-1.55" upper="1.55" velocity="0.005"/>
    <axis xyz="0 0 1"/>
  </joint>
  <transmission name="tran1">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="head_aux_1_joint">
      <hardwareInterface>EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="motor1">
      <hardwareInterface>EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <link name="aux_2_link">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0.0285"/>
      <mass value="0.01"/>
      <inertia ixx="2.64583333333e-08" ixy="0.0" ixz="0.0" iyy="2.64583333333e-08" iyz="0.0" izz="1.125e-08"/>
    </inertial>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0.00285"/>
      <geometry>
        <cylinder length="0.005" radius="0.0015"/>
      </geometry>
    </collision>
    <visual>
      <origin rpy="0.0 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder length="0.005" radius="0.0015"/>
      </geometry>
      <material name="blue"/>
    </visual>
  </link>
  <joint name="head_aux_2_joint" type="revolute">
    <parent link="head_link"/>
    <child link="aux_2_link"/>
    <origin rpy="3.145159 0 2.61799" xyz="0.02165 -0.0125 -0.008"/>
    <limit effort="1.0" lower="-1.55" upper="1.55" velocity="0.005"/>
    <axis xyz="0 0 1"/>
  </joint>
  <transmission name="tran2">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="head_aux_2_joint">
      <hardwareInterface>EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="motor2">
      <hardwareInterface>EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <link name="aux_3_link">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0.0285"/>
      <mass value="0.01"/>
      <inertia ixx="2.64583333333e-08" ixy="0.0" ixz="0.0" iyy="2.64583333333e-08" iyz="0.0" izz="1.125e-08"/>
    </inertial>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0.00285"/>
      <geometry>
        <cylinder length="0.005" radius="0.0015"/>
      </geometry>
    </collision>
    <visual>
      <origin rpy="0.0 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder length="0.005" radius="0.0015"/>
      </geometry>
      <material name="blue"/>
    </visual>
  </link>
  <joint name="head_aux_3_joint" type="revolute">
    <parent link="head_link"/>
    <child link="aux_3_link"/>
    <origin rpy="3.145159 0 0.523599" xyz="0 0.025 -0.008"/>
    <limit effort="1.0" lower="-1.55" upper="1.55" velocity="0.005"/>
    <axis xyz="0 0 1"/>
  </joint>
  <transmission name="tran3">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="head_aux_3_joint">
      <hardwareInterface>EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="motor3">
      <hardwareInterface>EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <link name="upperleg_M1_link">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0.03"/>
      <mass value="0.01"/>
      <inertia ixx="3.015625e-05" ixy="0.0" ixz="0.0" iyy="3.015625e-05" iyz="0.0" izz="3.125e-07"/>
    </inertial>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0.03"/>
      <geometry>
        <cylinder length="0.06" radius="0.0025"/>
      </geometry>
    </collision>
    <visual>
      <origin rpy="0.0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://gurdy_description/models/gurdy/meshes/gurdy_higherleg_v2.dae"/>
      </geometry>
    </visual>
  </link>
  <joint name="auxs_upperlegM1_joint" type="revolute">
    <parent link="aux_1_link"/>
    <child link="upperleg_M1_link"/>
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <limit effort="1.0" lower="-1.55" upper="0.0" velocity="0.005"/>
    <axis xyz="0 1 0"/>
  </joint>
  <transmission name="tran4">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="auxs_upperlegM1_joint">
      <hardwareInterface>EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="motor4">
      <hardwareInterface>EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <link name="upperleg_M2_link">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0.03"/>
      <mass value="0.01"/>
      <inertia ixx="3.015625e-05" ixy="0.0" ixz="0.0" iyy="3.015625e-05" iyz="0.0" izz="3.125e-07"/>
    </inertial>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0.03"/>
      <geometry>
        <cylinder length="0.06" radius="0.0025"/>
      </geometry>
    </collision>
    <visual>
      <origin rpy="0.0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://gurdy_description/models/gurdy/meshes/gurdy_higherleg_v2.dae"/>
      </geometry>
    </visual>
  </link>
  <joint name="auxs_upperlegM2_joint" type="revolute">
    <parent link="aux_2_link"/>
    <child link="upperleg_M2_link"/>
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <limit effort="1.0" lower="-1.55" upper="0.0" velocity="0.005"/>
    <axis xyz="0 1 0"/>
  </joint>
  <transmission name="tran5">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="auxs_upperlegM2_joint">
      <hardwareInterface>EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="motor5">
      <hardwareInterface>EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <link name="upperleg_M3_link">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0.03"/>
      <mass value="0.01"/>
      <inertia ixx="3.015625e-05" ixy="0.0" ixz="0.0" iyy="3.015625e-05" iyz="0.0" izz="3.125e-07"/>
    </inertial>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0.03"/>
      <geometry>
        <cylinder length="0.06" radius="0.0025"/>
      </geometry>
    </collision>
    <visual>
      <origin rpy="0.0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://gurdy_description/models/gurdy/meshes/gurdy_higherleg_v2.dae"/>
      </geometry>
    </visual>
  </link>
  <joint name="auxs_upperlegM3_joint" type="revolute">
    <parent link="aux_3_link"/>
    <child link="upperleg_M3_link"/>
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <limit effort="1.0" lower="-1.55" upper="0.0" velocity="0.005"/>
    <axis xyz="0 1 0"/>
  </joint>
  <transmission name="tran6">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="auxs_upperlegM3_joint">
      <hardwareInterface>EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="motor6">
      <hardwareInterface>EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <link name="lowerleg_M1_link">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0.03"/>
      <mass value="0.01"/>
      <inertia ixx="3.005625e-05" ixy="0.0" ixz="0.0" iyy="3.005625e-05" iyz="0.0" izz="1.125e-07"/>
    </inertial>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0.03"/>
      <geometry>
        <cylinder length="0.06" radius="0.0015"/>
      </geometry>
    </collision>
    <visual>
      <origin rpy="0.0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://gurdy_description/models/gurdy/meshes/gurdy_lowerleg_v2.dae"/>
      </geometry>
    </visual>
  </link>
  <joint name="upperlegM1_lowerlegM1_joint" type="continuous">
    <parent link="upperleg_M1_link"/>
    <child link="lowerleg_M1_link"/>
    <origin rpy="0 0 3.14159" xyz="0 0.0095 0.06"/>
    <limit effort="1.0" lower="-2.9" upper="1.5708" velocity="0.005"/>
    <axis xyz="0 1 0"/>
  </joint>
  <transmission name="tran7">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="upperlegM1_lowerlegM1_joint">
      <hardwareInterface>EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="motor7">
      <hardwareInterface>EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <link name="lowerleg_M2_link">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0.03"/>
      <mass value="0.01"/>
      <inertia ixx="3.005625e-05" ixy="0.0" ixz="0.0" iyy="3.005625e-05" iyz="0.0" izz="1.125e-07"/>
    </inertial>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0.03"/>
      <geometry>
        <cylinder length="0.06" radius="0.0015"/>
      </geometry>
    </collision>
    <visual>
      <origin rpy="0.0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://gurdy_description/models/gurdy/meshes/gurdy_lowerleg_v2.dae"/>
      </geometry>
    </visual>
  </link>
  <joint name="upperlegM2_lowerlegM2_joint" type="continuous">
    <parent link="upperleg_M2_link"/>
    <child link="lowerleg_M2_link"/>
    <origin rpy="0 0 3.14159" xyz="0 0.0095 0.06"/>
    <limit effort="1.0" lower="-2.9" upper="1.5708" velocity="0.005"/>
    <axis xyz="0 1 0"/>
  </joint>
  <transmission name="tran8">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="upperlegM2_lowerlegM2_joint">
      <hardwareInterface>EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="motor8">
      <hardwareInterface>EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <link name="lowerleg_M3_link">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0.03"/>
      <mass value="0.01"/>
      <inertia ixx="3.005625e-05" ixy="0.0" ixz="0.0" iyy="3.005625e-05" iyz="0.0" izz="1.125e-07"/>
    </inertial>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0.03"/>
      <geometry>
        <cylinder length="0.06" radius="0.0015"/>
      </geometry>
    </collision>
    <visual>
      <origin rpy="0.0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://gurdy_description/models/gurdy/meshes/gurdy_lowerleg_v2.dae"/>
      </geometry>
    </visual>
  </link>
  <joint name="upperlegM3_lowerlegM3_joint" type="continuous">
    <parent link="upperleg_M3_link"/>
    <child link="lowerleg_M3_link"/>
    <origin rpy="0 0 3.14159" xyz="0 0.0095 0.06"/>
    <limit effort="1.0" lower="-2.9" upper="1.5708" velocity="0.005"/>
    <axis xyz="0 1 0"/>
  </joint>
  <transmission name="tran9">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="upperlegM3_lowerlegM3_joint">
      <hardwareInterface>EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="motor9">
      <hardwareInterface>EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <link name="internallowerleg_M1_link">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0.03"/>
      <mass value="0.01"/>
      <inertia ixx="5.58333333333e-06" ixy="0.0" ixz="0.0" iyy="5.58333333333e-06" iyz="0.0" izz="5e-07"/>
    </inertial>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0.03"/>
      <geometry>
        <cylinder length="0.08" radius="0.001"/>
      </geometry>
    </collision>
    <visual>
      <origin rpy="0.0 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder length="0.03" radius="0.001"/>
      </geometry>
      <material name="blue"/>
    </visual>
  </link>
  <joint name="internal_lowerleg1_joint" type="prismatic">
    <parent link="lowerleg_M1_link"/>
    <child link="internallowerleg_M1_link"/>
    <origin rpy="0 0 0" xyz="0 0.0 0.06"/>
    <limit effort="1.0" lower="0" upper="0.02" velocity="0.005"/>
    <axis xyz="0 0 1"/>
  </joint>
  <transmission name="tran10">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="internal_lowerleg1_joint">
      <hardwareInterface>EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="motor10">
      <hardwareInterface>EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <link name="internallowerleg_M2_link">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0.03"/>
      <mass value="0.01"/>
      <inertia ixx="5.58333333333e-06" ixy="0.0" ixz="0.0" iyy="5.58333333333e-06" iyz="0.0" izz="5e-07"/>
    </inertial>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0.03"/>
      <geometry>
        <cylinder length="0.08" radius="0.001"/>
      </geometry>
    </collision>
    <visual>
      <origin rpy="0.0 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder length="0.03" radius="0.001"/>
      </geometry>
      <material name="blue"/>
    </visual>
  </link>
  <joint name="internal_lowerleg2_joint" type="prismatic">
    <parent link="lowerleg_M2_link"/>
    <child link="internallowerleg_M2_link"/>
    <origin rpy="0 0 0" xyz="0 0.0 0.06"/>
    <limit effort="1.0" lower="0" upper="0.02" velocity="0.005"/>
    <axis xyz="0 0 1"/>
  </joint>
  <transmission name="tran11">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="internal_lowerleg2_joint">
      <hardwareInterface>EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="motor11">
      <hardwareInterface>EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <link name="internallowerleg_M3_link">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0.03"/>
      <mass value="0.01"/>
      <inertia ixx="5.58333333333e-06" ixy="0.0" ixz="0.0" iyy="5.58333333333e-06" iyz="0.0" izz="5e-07"/>
    </inertial>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0.03"/>
      <geometry>
        <cylinder length="0.08" radius="0.001"/>
      </geometry>
    </collision>
    <visual>
      <origin rpy="0.0 0 0" xyz="0 0 0"/>
      <geometry>
        <cylinder length="0.03" radius="0.001"/>
      </geometry>
      <material name="blue"/>
    </visual>
  </link>
  <joint name="internal_lowerleg3_joint" type="prismatic">
    <parent link="lowerleg_M3_link"/>
    <child link="internallowerleg_M3_link"/>
    <origin rpy="0 0 0" xyz="0 0.0 0.06"/>
    <limit effort="1.0" lower="0" upper="0.02" velocity="0.005"/>
    <axis xyz="0 0 1"/>
  </joint>
  <transmission name="tran12">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="internal_lowerleg3_joint">
      <hardwareInterface>EffortJointInterface</hardwareInterface>
    </joint>
    <actuator name="motor12">
      <hardwareInterface>EffortJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <link name="footM1_link">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="0.01"/>
      <inertia ixx="1.28e-06" ixy="0.0" ixz="0.0" iyy="1.28e-06" iyz="0.0" izz="1.28e-06"/>
    </inertial>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <sphere radius="0.008"/>
      </geometry>
    </collision>
    <visual>
      <origin rpy="0.0 0 0" xyz="0 0 0"/>
      <geometry>
        <sphere radius="0.008"/>
      </geometry>
    </visual>
  </link>
  <gazebo reference="footM1_link">
    <kp>1000.0</kp>
    <kd>1000.0</kd>
    <mu1>10.0</mu1>
    <mu2>10.0</mu2>
    <material>Gazebo/Blue</material>
  </gazebo>
  <joint name="basefoot_peg_M1_joint" type="continuous">
    <parent link="internallowerleg_M1_link"/>
    <child link="footM1_link"/>
    <origin rpy="0 0 0" xyz="0 0 0.02"/>
  </joint>
  <link name="footM2_link">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="0.01"/>
      <inertia ixx="1.28e-06" ixy="0.0" ixz="0.0" iyy="1.28e-06" iyz="0.0" izz="1.28e-06"/>
    </inertial>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <sphere radius="0.008"/>
      </geometry>
    </collision>
    <visual>
      <origin rpy="0.0 0 0" xyz="0 0 0"/>
      <geometry>
        <sphere radius="0.008"/>
      </geometry>
    </visual>
  </link>
  <gazebo reference="footM2_link">
    <kp>1000.0</kp>
    <kd>1000.0</kd>
    <mu1>10.0</mu1>
    <mu2>10.0</mu2>
    <material>Gazebo/Blue</material>
  </gazebo>
  <joint name="basefoot_peg_M2_joint" type="continuous">
    <parent link="internallowerleg_M2_link"/>
    <child link="footM2_link"/>
    <origin rpy="0 0 0" xyz="0 0 0.02"/>
  </joint>
  <link name="footM3_link">
    <inertial>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <mass value="0.01"/>
      <inertia ixx="1.28e-06" ixy="0.0" ixz="0.0" iyy="1.28e-06" iyz="0.0" izz="1.28e-06"/>
    </inertial>
    <collision>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <sphere radius="0.008"/>
      </geometry>
    </collision>
    <visual>
      <origin rpy="0.0 0 0" xyz="0 0 0"/>
      <geometry>
        <sphere radius="0.008"/>
      </geometry>
    </visual>
  </link>
  <gazebo reference="footM3_link">
    <kp>1000.0</kp>
    <kd>1000.0</kd>
    <mu1>10.0</mu1>
    <mu2>10.0</mu2>
    <material>Gazebo/Green</material>
  </gazebo>
  <joint name="basefoot_peg_M3_joint" type="continuous">
    <parent link="internallowerleg_M3_link"/>
    <child link="footM3_link"/>
    <origin rpy="0 0 0" xyz="0 0 0.02"/>
  </joint>
  <!-- Including the sensors IMU & Camera -->
  <gazebo>
    <plugin filename="libgazebo_ros_imu.so" name="gazebo_ros_imu_controller">
      <robotNamespace>/gurdy</robotNamespace>
      <topicName>imu/data</topicName>
      <serviceName>imu/service</serviceName>
      <bodyName>base_link</bodyName>
      <gaussianNoise>0</gaussianNoise>
      <rpyOffsets>0 0 0</rpyOffsets>
      <updateRate>5.0</updateRate>
      <alwaysOn>true</alwaysOn>
      <gaussianNoise>0</gaussianNoise>
    </plugin>
  </gazebo>
  <gazebo reference="camera_link">
    <sensor name="camera1" type="camera">
      <update_rate>15.0</update_rate>
      <camera name="head">
        <pose>0 0 0.05 0 0.785 0</pose>
        <horizontal_fov>1.3962634</horizontal_fov>
        <image>
          <width>400</width>
          <height>400</height>
          <format>R8G8B8</format>
        </image>
        <clip>
          <near>0.01</near>
          <far>100</far>
        </clip>
        <noise>
          <type>gaussian</type>
          <stddev>0.007</stddev>
        </noise>
      </camera>
      <plugin filename="libgazebo_ros_camera.so" name="camera_controller">
        <alwaysOn>true</alwaysOn>
        <updateRate>0.0</updateRate>
        <cameraName>gurdy/camera1</cameraName>
        <imageTopicName>image_raw</imageTopicName>
        <cameraInfoTopicName>camera_info</cameraInfoTopicName>
        <frameName>camera_link</frameName>
        <hackBaseline>0.07</hackBaseline>
        <distortionK1>0.0</distortionK1>
        <distortionK2>0.0</distortionK2>
        <distortionK3>0.0</distortionK3>
        <distortionT1>0.0</distortionT1>
        <distortionT2>0.0</distortionT2>
      </plugin>
    </sensor>
  </gazebo>
  <!-- Differential drive controller  -->
  <gazebo>
    <plugin filename="libgazebo_ros_diff_drive.so" name="differential_drive_controller">
      <rosDebugLevel>Debug</rosDebugLevel>
      <publishWheelTF>false</publishWheelTF>
      <robotNamespace>/new_gurdy</robotNamespace>
      <publishTf>1</publishTf>
      <publishWheelJointState>false</publishWheelJointState>
      <alwaysOn>true</alwaysOn>
      <updateRate>100.0</updateRate>
      <leftJoint>basefoot_peg_M1_joint</leftJoint>
      <rightJoint>basefoot_peg_M2_joint</rightJoint>
      <wheelSeparation>0.1</wheelSeparation>
      <wheelDiameter>0.016</wheelDiameter>
      <broadcastTF>1</broadcastTF>
      <wheelTorque>30</wheelTorque>
      <wheelAcceleration>1.8</wheelAcceleration>
      <commandTopic>cmd_vel</commandTopic>
      <odometryFrame>odom</odometryFrame>
      <odometryTopic>odom</odometryTopic>
      <robotBaseFrame>base_link</robotBaseFrame>
      <legacyMode>false</legacyMode>
      <odometrySource>world</odometrySource>
      <!-- 'encoder' instead of 'world' is also possible -->
      <publishTf>1</publishTf>
    </plugin>
  </gazebo>
</robot>

